uiServices/src/testCaseServices.go

```diff
@@ -241,3 +241,38 @@ func getTestCase(rw http.ResponseWriter, req *http.Request) {
 
 }
 
+
+func deleteTestCase(rw http.ResponseWriter, req *http.Request){
+	testCasePathDir := getTestCaseHeader(req)
+	testCaseName := chi.URLParam(req, "testCaseName")
+	ValidateFileNameAndHeader(rw, req, testCasePathDir, testCaseName)
+
+	filepath := fmt.Sprintf("%s%s.xml", testCasePathDir, testCaseName)
+
+	if _, err := os.Stat(filepath); err != nil {
+		if os.IsNotExist(err) {
+			logrus.Println("File Not Found", err)
+			rw.WriteHeader(http.StatusNotFound)
+			return
+		}
+	}
+
+	err := os.Remove(filepath)
+	if  err != nil{
+		logrus.Println("File was not deleted", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+
+	}
+
+	if _, err := os.Stat(filepath); err != nil {
```

Comment:
Why do you need this block here?
If the file fails to get deleted you should get an error and return? Isn't that's what is happening on line 261-264?

Reply:
I added the extra check after to make sure it was definitely deleted. Just incase something happened where a person tried to delete a file that was opened by another user and it said file was deleted even though it may not have been. I will take it out.